interface conversion: interpreter.NilValue is not interpreter

### Instructions

Please fill out the template below to the best of your ability and include a label indicating which tool/service you were working with when you encountered the problem.

### Problem
DEBU[0010] ️✉️   Transaction submitted                   txID=bc0ff0694ada0e6f3026af39a59794001c92aa8731acc42e773a66031c4780da
panic: interface conversion: interpreter.NilValue is not interpreter.EquatableValue: missing method Equal [recovered]
	panic: interface conversion: interpreter.NilValue is not interpreter.EquatableValue: missing method Equal [recovered]



### Steps to Reproduce

Compile the contracts, and run set_account, then mint_nft.cdc
line 119, last line of my NFTMinter

### Acceptance Criteria

<if any>

### Context
I'm trying to have a private Collection, so when the user mints, it mints to my collection instead.
[testnet-dev.zip](https://github.com/onflow/cadence/files/6403197/testnet-dev.zip)


<what are you currently working on that this is blocking?>

--- a/runtime/tests/interpreter/equality_test.go
+++ b/runtime/tests/interpreter/equality_test.go
@@ -107,4 +107,19 @@ func TestInterpretEquality(t *testing.T) {
 			inter.Globals["res2"].GetValue(),
 		)
 	})
+
+	t.Run("nil", func(t *testing.T) {
+
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+          let n: Int? = 1
+          let res = nil == n
+		`)
+
+		assert.Equal(t,
+			interpreter.BoolValue(false),
+			inter.Globals["res"].GetValue(),
+		)
+	})
 }

--- a/runtime/tests/interpreter/switch_test.go
+++ b/runtime/tests/interpreter/switch_test.go
@@ -187,4 +187,74 @@ func TestInterpretSwitchStatement(t *testing.T) {
 			assert.Equal(t, expected, actual)
 		}
 	})
+
+	t.Run("optional", func(t *testing.T) {
+
+		inter := parseCheckAndInterpretWithOptions(t,
+			`
+              fun test(_ x: Int?, _ y: Int?): String {
+                  switch x {
+                  case y:
+                      return "1"
+                  case nil:
+                      return "2"
+                  default:
+                      return "3"
+                  }
+                  return "4"
+              }
+            `,
+			ParseCheckAndInterpretOptions{
+				HandleCheckerError: func(err error) {
+					errs := checker.ExpectCheckerErrors(t, err, 1)
+
+					assert.IsType(t, &sema.UnreachableStatementError{}, errs[0])
+				},
+			},
+		)
+
+		type testCase struct {
+			arguments []interpreter.Value
+			expected  interpreter.Value
+		}
+
+		for _, testCase := range []testCase{
+			{
+				[]interpreter.Value{
+					interpreter.NewSomeValueOwningNonCopying(
+						interpreter.NewIntValueFromInt64(1),
+					),
+					interpreter.NewSomeValueOwningNonCopying(
+						interpreter.NewIntValueFromInt64(1),
+					),
+				},
+				interpreter.NewStringValue("1"),
+			},
+			{
+				[]interpreter.Value{
+					interpreter.NilValue{},
+					interpreter.NewSomeValueOwningNonCopying(
+						interpreter.NewIntValueFromInt64(1),
+					),
+				},
+				interpreter.NewStringValue("2"),
+			},
+			{
+				[]interpreter.Value{
+					interpreter.NewSomeValueOwningNonCopying(
+						interpreter.NewIntValueFromInt64(1),
+					),
+					interpreter.NewSomeValueOwningNonCopying(
+						interpreter.NewIntValueFromInt64(2),
+					),
+				},
+				interpreter.NewStringValue("3"),
+			},
+		} {
+			actual, err := inter.Invoke("test", testCase.arguments...)
+			require.NoError(t, err)
+
+			assert.Equal(t, testCase.expected, actual)
+		}
+	})
 }

